components/net/http_loader.rs

```diff
@@ -1455,17 +1455,20 @@ fn cors_preflight_fetch(
     context: &FetchContext,
 ) -> Response {
     // Step 1
-    let mut preflight = Request::new(
-        request.current_url(),
-        Some(request.origin.clone()),
-        request.pipeline_id,
-    );
-    preflight.method = Method::OPTIONS;
-    preflight.initiator = request.initiator.clone();
-    preflight.destination = request.destination.clone();
-    preflight.origin = request.origin.clone();
-    preflight.referrer = request.referrer.clone();
-    preflight.referrer_policy = request.referrer_policy;
+    let mut preflight = RequestBuilder::new(request.current_url())
+        .method(Method::OPTIONS)
+        .origin(match request.origin {
+            Origin::Client => {
+                unimplemented!("Got client origin. Please check how we should handle it with request builder pattern")
```

Comment:
@jdm So, once we reach this point where we call `cors_preflight_fetch`, it is assumed that `Origin` here is never `Client`, which the spec says so [here](https://fetch.spec.whatwg.org/#concept-request-origin), and I verified that by looking at [step 1.2 of the fetch algorithm](https://fetch.spec.whatwg.org/#concept-fetch).

Reply:
In other words, we  can actually use `unreachable!()` here to say that this case is impossible to reach.

Reply:
Agreed!

Reply:
@KiChjang Thanks for the verification! That looks correct to me as well! Will update the PR! Thanks!!